src/windows/video_widget.py

```diff
@@ -1465,6 +1474,13 @@ def __init__(self, watch_project=True, *args):
         # Get a reference to the window object
         self.win = get_app().window
 
+        # Update title whenever playback speed changes.
+        self.win.PlaySignal.connect(self.update_title, Qt.QueuedConnection)
+        self.win.PlaySignal.connect(self.update_title, Qt.QueuedConnection)
+        self.win.PauseSignal.connect(self.update_title, Qt.QueuedConnection)
+        self.win.SpeedSignal.connect(self.update_title, Qt.QueuedConnection)
+        self.win.StopSignal.connect(self.update_title, Qt.QueuedConnection)
+
```

Comment:
I'd really like to avoid adding all these signal connections to the preview widget _just_ to add the speed to its title, since they're otherwise unnecessary.

As an alternative, my #4697 (which I've just updated to clean up some of the tangential stuff) moves the code that sets the preview title into the `MainWindow` class, which would make these connections unnecessary. (I would argue that code belongs there anyway, as the preview widget doesn't know anything about the dock it lives in. That's part of the child hierarchy for MainWindow, its owner.) 

Adding in speed display when the code lives in MainWindow is a snap, since there's already code there to keep track of the playback speed. (I don't care about the formatting of the stats display, I can change it to match the format here if that's preferred.)

Reply:
I have no issues with these signal connections inside the widget. However, I'm also not opposed to #4697, which looks very nice as well. Sorry, a little late with my feedback on this one.